src/main/java/frc/team2412/robot/subsystem/ClimbSubsystem.java

```diff
@@ -112,24 +162,37 @@ public void stopAngledArm() {
 
     public void setMotor(double position, WPI_TalonFX motor) {
         if (state == ClimbSubsystemState.ENABLED) {
+            System.out.println("motor set: " + position);
             motor.set(ControlMode.Position, position);
+            // motor.set(ControlMode.PercentOutput, 1); simulator only
         }
     }
 
     public boolean isFixedFullyExtended() {
-        return climbFixedMotor.getSelectedSensorPosition() >= ClimbConstants.MAX_ENCODER_TICKS;
+        return climbFixedMotor.getSelectedSensorPosition() >= maxEncoderTicks.getValue().getDouble();
     }
 
     public boolean isFixedFullyRetracted() {
-        return climbFixedMotor.getSelectedSensorPosition() <= ClimbConstants.MIN_ENCODER_TICKS;
+        return climbFixedMotor.getSelectedSensorPosition() <= minEncoderTicks.getValue().getDouble();
     }
 
     public boolean isDynamicFullyExtended() {
-        return climbDynamicMotor.getSelectedSensorPosition() >= ClimbConstants.MAX_ENCODER_TICKS;
+        return climbDynamicMotor.getSelectedSensorPosition() >= maxEncoderTicks.getValue().getDouble();
     }
 
     public boolean isDynamicFullyRetracted() {
-        return climbDynamicMotor.getSelectedSensorPosition() <= ClimbConstants.MIN_ENCODER_TICKS;
+        return climbDynamicMotor.getSelectedSensorPosition() <= minEncoderTicks.getValue().getDouble();
     }
 
+    @Override
+    public void simulationPeriodic() {
+        // max speed 6000 rpm, 2048 ticks per rotation
+        double timeNow = Timer.getFPGATimestamp();
+        double timeElapsed = timeNow - lastUpdatedTime;
+        double motorFixedSpeed = climbFixedMotor.getSelectedSensorVelocity();
+        double motorDynamicSpeed = climbDynamicMotor.getSelectedSensorVelocity();
```

Comment:
The sensor velocity is likely zero since it's not set above (`setMotor` sets the position). You could have `setMotor` check `Robot.isSimulation()` and if in simulation mode, call .getSimCollection().setIntegratedSensorVelocity(<something>) to set the velocity to something (in units of ticks per 100ms).